### ISSUE TITLE
Mypy does not report the expected signature upon mismatch with the supertype's

### ISSUE THREAD

**Issue by nbraud** [python/mypy#10983]
**Bug Report**

When deriving from a (generic) class, such as `Sequence[int]`, mypy may report that the signature of a given method is incompatible with supertype's, but without displaying the supertype's.

This is extremely problematic, in cases like `Sequence.__getitem__`, where the expected signature does not appear to be listed in either the documentation or code, of `typing` or `collections.abc`.

**To Reproduce**

Typecheck the following:

```py3
from collections.abc import Sequence

class SmolSet(Sequence[int]):
    __slots__ = ("x",)
    x: int

    def __init__(self):
        self.x = 0

    def __len__(self) -> int:
        return 32

    def __getitem__(self, item: int) -> int:
        if not 0 <= item < 32:
            raise IndexError(f"{type(self).__name__} "
                             "expects indices between 0 and 32 (exclusive)")

        return (self.x >> (2 * item)) & 3
```

**Expected Behavior**

Either typecheck successfully, or provide actionable feedback on what the issue is.

**Actual Behavior**

```
$ mypy smolsets.py
smolsets.py:13: error: Signature of "__getitem__" incompatible with supertype "Sequence"
Found 1 error in 1 file (checked 1 source file)
```

Then spend an hour fruitlessly searching for what is (mypy's idea of) `Sequence.__getitem__`'s signature.

**Your Environment**

- Mypy version used: 0.812
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.9.6
- Operating system and version: NixOS unstable

**Comment by JelleZijlstra** [comment]
Mypy version 0.812 is quite old. We made a recent change to improve this similar to your request, but I'm not sure that change has been released yet.

**Comment by nbraud** [comment]
@JelleZjistra Yes, I tried upgrading mypy in NixOS (the distribution I use) but ran into an unrelated issue trying to package it; I should try installing the latest version from PyPI and see whether that helps.

**Comment by nbraud** [comment]
@JelleZijlstra You were right, newer `mypy` versions do provide useful feedback there.
